packages/components/src/styles/common-styles.ts

```diff
@@ -5,18 +5,50 @@ import type { ThemedColors } from './';
 import {
   borderWidthBase,
   frostedGlassStyle,
+  motionDurationModerate,
+  motionDurationShort,
+  motionDurationLong,
+  motionDurationVeryLong,
+  motionEasingBase,
+  motionEasingIn,
+  motionEasingOut,
   themeDarkBackgroundShading,
   themeLightBackgroundShading,
   themeLightStateFocus,
 } from '@porsche-design-system/utilities-v2';
 import { getThemedColors, prefersColorSchemeDarkMediaQuery } from './';
 import { isThemeDark } from '../utils';
+import type * as fromMotion from '@porsche-design-system/utilities-v2/dist/esm/motion';
 
-export const transitionDuration = 'var(--p-transition-duration, .24s)';
-const transitionTimingFunction = 'ease';
+type WithoutMotionDurationPrefix<T> = T extends `motionDuration${infer P}` ? Uncapitalize<P> : never;
+export type MotionDurationKeyFinal = WithoutMotionDurationPrefix<keyof typeof fromMotion>;
+type WithoutMotionEasingPrefix<T> = T extends `motionEasing${infer P}` ? Uncapitalize<P> : never;
+export type MotionEasingKeyFinal = WithoutMotionEasingPrefix<keyof typeof fromMotion>;
 
-export const getTransition = (cssProperty: keyof PropertiesHyphen): string =>
-  `${cssProperty} ${transitionDuration} ${transitionTimingFunction}`;
+const motionDurationMap: Record<MotionDurationKeyFinal, string> = {
+  short: motionDurationShort,
+  moderate: motionDurationModerate,
+  long: motionDurationLong,
+  veryLong: motionDurationVeryLong,
+};
+
+const motionEasingMap: Record<MotionEasingKeyFinal, string> = {
+  base: motionEasingBase,
+  in: motionEasingIn,
+  out: motionEasingOut,
+};
+
+export const cssVariableMotionDuration = '--p-motion-duration';
+
+export const getTransition = (
+  cssProperty: keyof PropertiesHyphen,
+  duration: MotionDurationKeyFinal | '0s' = 'short',
+  easing: MotionEasingKeyFinal | 'linear' | 'none' = 'base',
+  delay?: MotionDurationKeyFinal | '0s'
+): string =>
+  `${cssProperty} var(${cssVariableMotionDuration}, ${duration === '0s' ? duration : motionDurationMap[duration]})${
+    easing === 'none' ? '' : ` ${easing === 'linear' ? easing : motionEasingMap[easing]}`
+  }${delay ? ` var(${cssVariableMotionDuration}, ${delay === '0s' ? delay : motionDurationMap[delay]})` : ''}`;
```

Comment:
First `delay ?` check will always be true with the default value?

Reply:
delay does not have a default value